sdk/cosmos/azure-cosmos/azure/cosmos/_change_feed/aio/change_feed_iterable.py

```diff
@@ -0,0 +1,166 @@
+# The MIT License (MIT)
+# Copyright (c) 2014 Microsoft Corporation
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Iterable change feed results in the Azure Cosmos database service.
+"""
+from typing import Dict, Any
+
+from azure.core.async_paging import AsyncPageIterator
+
+from azure.cosmos import PartitionKey
+from azure.cosmos._change_feed.aio.change_feed_fetcher import ChangeFeedFetcherV1, ChangeFeedFetcherV2
+from azure.cosmos._change_feed.change_feed_state import ChangeFeedState, ChangeFeedStateV1
+from azure.cosmos._utils import is_base64_encoded, is_key_exists_and_not_none
+
+# pylint: disable=protected-access
+
+class ChangeFeedIterable(AsyncPageIterator):
+    """Represents an iterable object of the change feed results.
+
+    ChangeFeedIterable is a wrapper for change feed execution.
+    """
+
+    def __init__(
+        self,
+        client,
+        options,
+        fetch_function=None,
+        collection_link=None,
+        continuation_token=None,
+    ):
+        """Instantiates a ChangeFeedIterable for non-client side partitioning queries.
+
+        ChangeFeedFetcher will be used as the internal query execution
+        context.
+
+        :param CosmosClient client: Instance of document client.
+        :param dict options: The request options for the request.
+        :param method fetch_function:
```

Comment:
nit: `collection_link` and `continuation_token` related stuff can be specified here as well.

Reply:
added